lab.js

```diff
@@ -0,0 +1,118 @@
+// # Recursion Exercises
+
+// - ### Sum of all from 1 to n
+
+// Write a function that takes in a number as input and recursively finds the sum of all numbers up to and including that number.
+
+// ```js
+// input: 6
+// output: 21
+
+// //21 = 6 + 5 + 4 + 3 + 2 + 1
+// ```
+
+const sumNum = (num) => {
+    if(num === 0) return 0;
+    return num + sumNum(num - 1)
+}
+
+// console.log(sumNum(6))
+
+// - ### Multiply array
+
+// Write a function called `multArr` that takes in an array of numbers as an argument and recursively multiplies together all of the values in the array.
+
+// ```js
+// multArr([2, 3, 5]); // returns 30
+// multArr([5, 5, 1, 2]); //returns 50
+// ```
+
+const multArr = (arr) => {
+    if (arr.length === 0) return 1;
+    let el = arr.pop()
+    return multArr(arr) * el
+}
+console.log(multArr([2,3,5]))
+
+// - ### Concatenate array
+
+// Write a function called `concatArr` that takes in an array of strings as an argument and recursively concatenates the strings together into a single string, with spaces between each original string.
+
+// ```js
+// concatArr(['is', 'it', 'tomorrow']); // returns 'is it tomorrow'
+// concatArr(['or', 'just', 'the', 'end', 'of', 'time']); //returns 'or just the end of time'
+// ```
+
+const concatArr = (arr) => {
+    if(arr.length === 0) return "";
+    let joinStr = arr.pop()
+    return concatArr(arr) + joinStr + " "
+}
+
+console.log(concatArr(['is', 'it', 'tomorrow']))
+// - ### Sum evens
+
+// Write a function called `sumEvens` that takes in an array of numbers as an argument and recursively sums only the even numbers in the array.
+
+// ```js
+// sumEvens([2, 3, 5, 6]); // returns 8
+// sumEvens([10, 5, 1, 2, 12]); //returns 24
+// ```
+
+// const sumEvens = (arr) => {
+//     if (arr.length === 0) return;
+//     let el = arr.pop()
+//     if (el % 2 === 0) {
+//         return sumEvens(arr) + el
+//     } else {
+//         return 0
```

Comment:
Your else statement shouldn't return 0, it should call the function sumEvens otherwise your program will stop even if you haven't process all your array